internal/pkg/db/mongo/metadata.go

```diff
@@ -931,26 +898,30 @@ func (mc MongoClient) getCommandById(id string) (models.Command, error) {
 	s := mc.session.Copy()
 	defer s.Close()
 
-	query, err := idToBsonM(id)
-	if err != nil {
-		return models.Command{}, err
-	}
-
-	var command models.Command
-	if err = s.DB(mc.database.Name).C(db.Command).Find(query).One(&command); err != nil {
+	query := bson.M{"commands.uuid": id}
+	var dp models.DeviceProfile
+	if err := s.DB(mc.database.Name).C(db.DeviceProfile).Find(query).Select(bson.M{"commands": 1}).One(&dp); err != nil {
 		return models.Command{}, errorMap(err)
 	}
-	return command, nil
+	if len(dp.CoreCommands) > 1 {
+		return models.Command{}, fmt.Errorf("not unique result found for command with %v", id)
```

Comment:
Testing locally, the function is returning from this statement, which then returns a 500.
`GET http://localhost:48081/api/v1/command/ab29bd48-8ce5-4e74-bc5c-0e27031d4d9b`

In order to test I've started all of the edgex-go services and then also started the device-random service in order to populate deviceprofile, device, commands, etc. So that command ID is coming from device-random. When I look at the raw JSON of the device profile, the command ID `ab29bd48-8ce5-4e74-bc5c-0e27031d4d9b` is unique, so something about the above query isn't working.

Blackbox tests also fail
<img width="577" alt="Screen Shot 2019-06-04 at 4 19 03 PM" src="https://user-images.githubusercontent.com/1342980/58910821-81365d80-86e4-11e9-87bd-bf059358be5e.png">

Reply:
I have reproduced it from the first time. Thanks
Probably, I have tested it with multiple deviceProfiles that contain only single command .....hm